mitiq/interface/mitiq_qibo/conversions.py

```diff
@@ -0,0 +1,531 @@
+        gates.RZ(q1, theta),
+        gates.CNOT(q0, q1),
+    ]
+    return decomp_gate
+
+
+GATES_TO_DECOMPOSE = {
+    "crx": crx_decomp,
+    "cry": cry_decomp,
+    "crz": crz_decomp,
+    "cu1": cu1_decomp,
+    "cu3": cu3_decomp,
+    "iswap": iswap_decomp,
+    "fswap": fswap_decomp,
+    "rxx": rxx_decomp,
+    "ryy": ryy_decomp,
+    "rzz": rzz_decomp,
+}
+
+
+class UnsupportedCirqCircuitError(Exception):
+    pass
+
+
+class UnsupportedQiboCircuitError(Exception):
+    pass
+
+
+def decompose_qibo_circuit(qibo_circuit: QiboCircuit) -> QiboCircuit:
+    """Returns a QiboCircuit circuit equivalent to the input QiboCircuit
+    with all unknown by cirq gates decomposed.
+
+    Args:
+        qibo_circuit: QiboCircuit to decompose.
+
+    Returns:
+        Decomposed QiboCircuit
+    """
+    decomp_circuit = QiboCircuit(qibo_circuit.nqubits)
+    for gate in qibo_circuit.queue:
+        if gate.name in GATES_TO_DECOMPOSE:
+            function = GATES_TO_DECOMPOSE[gate.name]
+            decomposed_gate = function(gate)
+            decomp_circuit.add(decomposed_gate)
+        else:
+            decomp_circuit.add(gate)
+
+    return decomp_circuit
+
+
+def from_qibo(qibo_circuit: QiboCircuit) -> Circuit:
+    """Returns a Cirq circuit equivalent to the input QiboCircuit.
+
+    Args:
+        qibo_circuit: QiboCircuit to convert to a Cirq circuit.
+
+    Returns:
+        Cirq circuit representation equivalent to the input QiboCircuit.
+    """
+    for measurement in qibo_circuit.measurements:
+        reg_name = measurement.register_name
+        reg_qubits = measurement.target_qubits
+        if not reg_name.islower():
+            raise UnsupportedQiboCircuitError(
+                f"OpenQASM does not support capital letters in register names but {reg_name} was used."
+            )
+    for gate in qibo_circuit.queue:
+        if isinstance(gate, gates.M):
+            continue
+        if gate.is_controlled_by:
+            raise UnsupportedQiboCircuitError(
+                "OpenQASM does not support multi-controlled gates."
+            )
+        if gate.name not in gates.QASM_GATES:
+            raise UnsupportedQiboCircuitError(
+                f"{gate.name} is not supported by OpenQASM."
+            )
+
+    # Decompose the circuit in known Cirq gates
+    qasm = decompose_qibo_circuit(qibo_circuit).to_qasm()
+
+    return cirq_from_qasm(qasm)
+
+
+def to_qibo(circuit: Circuit) -> QiboCircuit:
+    """Returns a QiboCircuit equivalent to the input Cirq circuit.
+
+    Args:
+        circuit: Cirq circuit to convert to a QiboCircuit.
+
+    Returns:
+        QiboCircuit object equivalent to the input Mitiq circuit.
+    """
+    
+    try:  
+        qasm = cirq_to_qasm(circuit)
+        nqubits, gate_list = _parse_qasm_modified(qasm)
+        
+    except ValueError: 
+        decomp_circuit = Circuit(decompose(circuit)) 
+        qasm = cirq_to_qasm(decomp_circuit)
+        nqubits, gate_list = _parse_qasm_modified(qasm)
```

Comment:
Could we use the [`qibo.models.circuit.Circuit.from_qasm`](https://github.com/qiboteam/qibo/blob/7379f86437b448cbd5e93abd8129d0fa1b022ddb/src/qibo/models/circuit.py#L1047)
 function here? It would allow us to remove a lot of the bespoke code you wrote later in the file.


```suggestion
    try:  
        qasm = cirq_to_qasm(circuit)
    except ValueError: 
        decomp_circuit = Circuit(decompose(circuit)) 
        qasm = cirq_to_qasm(decomp_circuit)
    return QiboCircuit.from_qasm(qasm)
```

Reply:
I tested this change and it works!

Reply:
The issue with using `qibo.models.circuit.Circuit.from_qasm` with Qibo version 0.1.11 is that, in this version, the gates `sx` and` sxdg `are not defined. This causes the benchmark tests, specifically `test_mirror_circuits.py` and `test_mirror_qv_circuits.py`, to fail. This is why I had to include a modified version of the` from_qasm` function from Qibo in the code, which has led to the necessity of adding a significant amount of code.

Reply:
Ahhh okay I was failing to have run the benchmark tests. Sorry about the false start there then! Hopefully we can get the latest openfermion as you've been trying and can go from there.